File: C-Programming/Competitive/Assignment_39_3.c
#include <stdio.h>
#include <stdlib.h>

struct node
{
   int data;
   struct node *next;
};

typedef struct node NODE;
typedef struct node * PNODE;
typedef struct node ** PPNODE;

int CountOdd(PNODE first)
{
   int iCount = 0;
   while (first != NULL)
   {
      if(first->data % 2 != 0)
      {
         iCount++;
      }
      first = first->next;
   }
   return iCount;
}

void InsertFirst(PPNODE first, int iNo)
{
   PNODE newn = NULL;

   newn = (PNODE)malloc(sizeof(NODE));

   newn->data = iNo;
   newn->next = NULL;

   if(*first == NULL)
   {
      *first = newn;
   }
   else
   {
      newn->next = *first;
      *first = newn;
   }
}

int main()
{
   PNODE head = NULL;
   int iRet = 0;

   InsertFirst(&head,122);
   InsertFirst(&head,102);
   InsertFirst(&head,51);
   InsertFirst(&head,21);
   InsertFirst(&head,11);

   iRet = CountOdd(head);
   printf("Odd numbers are: %d", iRet);

   return 0;
}

File: Data-Structure/Assignment_39_3.java
import java.util.*;

class Node
{
   public int data;
   public Node next;
   
   public Node(int Value)
   {
      this.data = Value;
      this.next = null;
   }
}

class SinglyLinkedList
{
   private Node first;
   
   public SinglyLinkedList()
   {
      this.first = null;
   }
   
   public void InsertLast(int iNo)
   {
      Node newn = new Node(iNo);
      if(first == null)
      {
         first = newn;
      }
      else
      {
         Node temp = first;
         while(temp.next != null)
         {
            temp = temp.next;
         }
         temp.next = newn;
      }
   }

   public void Display()
   {
      Node temp = first;
      while(temp != null)
      {
         System.out.print("| " + temp.data + " | -> ");
         temp = temp.next;
      }
      System.out.println("NULL");
   }

   public int CountOddNumbers()
   {
      int iCount = 0;
      Node temp = first;
      while(temp != null)
      {
         if(temp.data % 2 != 0)
         {
            iCount++;
         }
         temp = temp.next;
      }
      return iCount;
   }
}

class Assignment_39_3
{
   public static void main(String A[])
   {
      int iRet = 0;
      SinglyLinkedList ssobj = new SinglyLinkedList();
      ssobj.InsertLast(21);
      ssobj.InsertLast(11);
      ssobj.InsertLast(51);
      ssobj.InsertLast(101);
      ssobj.InsertLast(111);

      ssobj.Display();

      iRet = ssobj.CountOddNumbers();
      
      System.out.println("Count of odd elements in the list is : " + iRet);
   }
}